### ISSUE TITLE
slf4j LoggerFactory ClassNotFound in Chapter 4

### ISSUE THREAD

**Issue by junglie85** [denisftw/modern-web-kotlin#3]
Great work on the book, finding it useful so far. I thought it worth pointing out that there seems to be an omission in Chapter 4. I couldn't get the example (on page 53) introducing Thymeleaf to compile without adding a compile dependency on org.slf4j:slf4j-api:1.7.25. I note in your build.gradle you have this dependency, although version 1.7.21.

I hope this will be of use for improving any future editions you author, and also of help to anyone else who has the same issue and isn't sure on the solution.

**Comment by denisftw** [comment]
Yeah, it's difficult to find the source of the problem as many Java libraries list SLF4J API as a dependency. Apparently, the previous version of Vert.x (the one that I'm using in the book) had such a dependency, so the necessary JAR was always on classpath. It's still not clear why it didn't complain about the absence of implementation at runtime, though.

I think that the best solution is to manually add the `slf4j-api` and `slf4j-jdk14` libraries to the Gradle build file to allow Vert.x to use its logging. Then, in the deployment chapter, we'll replace `slf4j-jdk14` with Logback and add `logback.xml` with proper configuration as well.

Thanks again for spotting it!
